src/hotspot/share/opto/compile.hpp

```diff
@@ -787,6 +788,9 @@ class Compile : public Phase {
   void   set_post_loop_opts_phase() { _post_loop_opts_phase = true;  }
   void reset_post_loop_opts_phase() { _post_loop_opts_phase = false; }
 
+  bool       allow_macro_nodes() { return _allow_macro_nodes;  }
+  void  dont_allow_macro_nodes() { _allow_macro_nodes = false;  }
```

Comment:
`dont_allow_macro_nodes` is a confusing name here, especially given it is a setter in contrast to `allow_macro_nodes`. Let's call it `reset_allow_macro_nodes()`.